src/main/scala/beam/utils/csv/conversion/XmlConverter.scala

```diff
@@ -0,0 +1,114 @@
+//  convert(householdsXml, new HouseholdsXml2CsvConverter(houseHoldAttributesXml).toCsv)
+//  convert(populationXml, PlansXml2CsvConverter.toCsv, newName = Some(new File("test/input/beamville/plans")))
+
+  def generatePopulationCsv(beamConfig: BeamConfig, allFiles: Seq[File]): File = {
+    val populationXml = new File(beamConfig.beam.agentsim.agents.plans.inputPlansFilePath)
+    val householdsXml: File = extractFile(
+      allFiles
+        .find(_.getAbsolutePath.endsWith("Households.xml.gz"))
+        .getOrElse(throw new IllegalStateException("could not find Households.xml.gz"))
+    )
+    val populationAttributesXml: File = extractFile(
+      allFiles
+        .find(_.getAbsolutePath.endsWith("PersonAttributes.xml.gz"))
+        .getOrElse(throw new IllegalStateException("could not find PersonAttributes.xml.gz"))
+    )
+    convert(
+      populationXml,
+      new PopulationXml2CsvConverter(householdsXml, populationAttributesXml).toCsv,
+      Some(new File(allFiles.head.getParentFile + "/population"))
+    )
+  }
+
+  def generateHouseholdsCsv(beamConfig: BeamConfig, allFiles: Seq[File]): File = {
+    val householdsXml = extractFile(
+      allFiles
+        .find(_.getAbsolutePath.endsWith("Households.xml.gz"))
+        .getOrElse(throw new IllegalStateException("could not find PersonAttributes.xml.gz"))
+    )
+    val houseHoldAttributesXml = new File(beamConfig.beam.agentsim.agents.households.inputHouseholdAttributesFilePath)
+    convert(
+      householdsXml,
+      new HouseholdsXml2CsvConverter(houseHoldAttributesXml).toCsv,
+      Some(new File(allFiles.head.getParentFile + "/households"))
+    )
+  }
+
+  def generatePlansCsv(beamConfig: BeamConfig, allFiles: Seq[File]): File = {
+    val populationXml = new File(beamConfig.beam.agentsim.agents.plans.inputPlansFilePath)
+    convert(
+      populationXml,
+      PlansXml2CsvConverter.toCsv,
+      Some(new File(allFiles.head.getParentFile + "/plans"))
+    )
+  }
+
+  def generateVehiclesCsv(beamConfig: BeamConfig, allFiles: Seq[File]): File = {
+    val file = new File(beamConfig.beam.agentsim.agents.vehicles.vehiclesFilePath)
+    val stream = new FileInputStream(file)
+    val newFile = new File(allFiles.headOption.get.getParentFile + "/vehicles.csv")
+    println(s"Generating file $newFile")
+    Files.copy(stream, newFile.toPath)
+    stream.close()
+    newFile
+  }
+
+  def csvFileName(file: File): File = new File(file.getAbsolutePath.stripSuffix(".xml") + ".csv")
+
+  def convert(file: File, f: File => Iterator[String], newName: Option[File] = None): File = {
+    val csvFile = newName.orElse(Some(file)).map(csvFileName).get
+    println(s"Generating file $csvFile")
+    FileUtils.writeToFile(
+      csvFile.getAbsolutePath,
+      f(file)
+    )
+    csvFile
+  }
+
+  def extractFile(gzipFile: File): File = {
+    val result = File.createTempFile("somePrefix", ".xml")
+    val outputStream = new FileOutputStream(result)
```

Comment:
If exception happens, the stream will remain open. Better to use `try catch`, or even better to use `IOUtils.getBufferedReader` which knows about gzipped file and create GZipStream on fly, example of usage [NetworkXmlToCSV.scala#L19](https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/scala/beam/utils/NetworkXmlToCSV.scala#L19)